pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerBase.java

```diff
@@ -895,24 +897,28 @@ private void pendingBatchReceiveTask(Timeout timeout) throws Exception {
     protected void triggerListener() {
         // Trigger the notification on the message listener in a separate thread to avoid blocking the networking
         // thread while the message processing happens
-        Message<T> msg;
-        do {
-            try {
-                msg = internalReceive(0, TimeUnit.MILLISECONDS);
+        try {
+            while (executorQueueSize.get() < conf.getReceiverQueueSize()) {
```

Comment:
Looks zero queue consumer could not receive messages?

Reply:
ZeroQueueConsumerImpl has override the triggerListener:
```java
    @Override
    protected void triggerListener() {
        // Ignore since it was already triggered in the triggerZeroQueueSizeListener() call
    }
```